aten/src/ATen/native/quantized/cpu/qclamp.cpp

```diff
@@ -14,12 +17,74 @@ namespace native {
 DEFINE_DISPATCH(qclamp_stub);
 
 namespace {
+
+#ifdef USE_PYTORCH_QNNPACK
+Tensor qnnpack_clamp(Tensor input, Scalar min, Scalar max) {
+
+  TORCH_CHECK(input.ndimension() > 0, "qnnpack_clamp(): Got empty input tensor");
+
+  initQNNPACK();
+
+  Tensor input_contig = input.contiguous();
+  size_t num_elems = input_contig.numel() / input_contig.size(0);
+
+  auto min_f = min.to<float>();
+  auto max_f = max.to<float>();
+  uint8_t min_q =
+      at::quantize_val<quint8>(input.q_scale(), input.q_zero_point(), min_f).val_;
+  uint8_t max_q =
+      at::quantize_val<quint8>(input.q_scale(), input.q_zero_point(), max_f).val_;
+
+  pytorch_qnnp_operator_t clamp_op{nullptr};
+  const pytorch_qnnp_status createStatus = pytorch_qnnp_create_clamp_nc_u8(
+    num_elems, // channels
+    min_q,
+    max_q,
+    0, // flags
+    &clamp_op);
+  TORCH_INTERNAL_ASSERT(createStatus == pytorch_qnnp_status_success,
+                        "failed to create QNNPACK Clamp operator");
+
+  Tensor qy = at::_empty_affine_quantized(
+    input_contig.sizes(),
+    input.options(),
```

Comment:
Same here, can you specify the input memory format?

Reply:
looks like it's already coming through with the options